Author: Mathieu Malaterre <malat@debian.org>
Last-Update: Wed Nov 27 10:33:27 2013 +0100
Bug-Debian: http://bugs.debian.org/730612
Description: Fix a compilation error with clang

--- a/processline.h
+++ b/processline.h
@@ -144,7 +144,7 @@ class ProcessTransformed : public Proces
 {
 	typedef typename TRANSFORM::SAMPLE SAMPLE;
 
-	ProcessTransformed(const ProcessTransformed&) {}
+	ProcessTransformed(const ProcessTransformed&);
 public:
 	ProcessTransformed(void* pbyteOutput, const JlsParameters& info, TRANSFORM transform) :
 		_pbyteOutput((BYTE*)pbyteOutput),
