Description: Replace deprecated FFmpeg API
Author: Andreas Cadhalpun <Andreas.Cadhalpun@googlemail.com>
Last-Update: <2015-11-02>

--- amide-1.0.5.orig/src/mpeg_encode.c
+++ amide-1.0.5/src/mpeg_encode.c
@@ -301,7 +301,7 @@ gpointer mpeg_encode_setup(gchar * outpu
   encode->context->time_base= (AVRational){1,FRAMES_PER_SECOND};
   encode->context->gop_size = 10; /* emit one intra frame every ten frames */
   encode->context->max_b_frames=10;
-  encode->context->pix_fmt = PIX_FMT_YUV420P;
+  encode->context->pix_fmt = AV_PIX_FMT_YUV420P;
 
   /* encoding parameters */
   encode->context->sample_aspect_ratio= (AVRational){1,1}; /* our pixels are square */
