diff --git a/eric6-6.0.4/eric/eric6.desktop b/eric6-6.0.4/eric/eric6.desktop
index c24db3f..2b00e72 100644
--- a/eric6-6.0.4/eric/eric6.desktop
+++ b/eric6-6.0.4/eric/eric6.desktop
@@ -3,7 +3,7 @@ Version=1.0
 Type=Application
 Exec=eric6@MARKER@
 MimeType=text/x-python;
-Icon=eric@MARKER@
+Icon=eric6@MARKER@
 Terminal=false
 Name=eric6@PY_MARKER@
 Name[de]=eric6@PY_MARKER@
diff --git a/eric6-6.0.4/eric/eric6_webbrowser.desktop b/eric6-6.0.4/eric/eric6_webbrowser.desktop
index 9a43eb0..82e6ca3 100644
--- a/eric6-6.0.4/eric/eric6_webbrowser.desktop
+++ b/eric6-6.0.4/eric/eric6_webbrowser.desktop
@@ -3,7 +3,7 @@ Version=1.0
 Type=Application
 Exec=eric6_webbrowser@MARKER@
 MimeType=text/html;text/xml;application/xhtml+xml;x-scheme-handler/http;x-scheme-handler/https;
-Icon=ericWeb@MARKER@
+Icon=eric6Web@MARKER@
 Terminal=false
 Name=eric6 Web Browser@PY_MARKER@
 Name[de]=eric6 Web Browser@PY_MARKER@
diff --git a/eric6-6.0.4/install.py b/eric6-6.0.4/install.py
index fadf244..8e11af3 100644
--- a/eric6-6.0.4/install.py
+++ b/eric6-6.0.4/install.py
@@ -787,10 +787,10 @@ def installEric():
                 os.makedirs(dst)
             shutilCopy(
                 os.path.join(sourceDir, "icons", "default", "eric6.png"),
-                os.path.join(dst, "eric" + marker + ".png"))
+                os.path.join(dst, "eric6" + marker + ".png"))
             shutilCopy(
                 os.path.join(sourceDir, "icons", "default", "ericWeb48.png"),
-                os.path.join(dst, "ericWeb" + marker + ".png"))
+                os.path.join(dst, "eric6Web" + marker + ".png"))
             dst = os.path.normpath(
                 os.path.join(distDir, "usr/share/applications"))
             if not os.path.exists(dst):
