diff -ru dcmtk-3.6.2.old/CMake/dcmtkPrepare.cmake dcmtk-3.6.2/CMake/dcmtkPrepare.cmake
--- dcmtk-3.6.2.old/CMake/dcmtkPrepare.cmake	2017-11-15 23:48:06.675985377 +0100
+++ dcmtk-3.6.2/CMake/dcmtkPrepare.cmake	2017-11-15 23:51:16.003980858 +0100
@@ -417,7 +417,8 @@
 
 # determine which flags are required to enable C++11 features (if any)
 IF(NOT DEFINED DCMTK_CXX11_FLAGS)
-  IF(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+  IF((CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 6.0.0)
+      OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
     SET(DCMTK_CXX11_FLAGS "-std=c++11")
   ELSEIF(CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
     IF(CMAKE_HOST_WIN32)
Nur in dcmtk-3.6.2/CMake: dcmtkPrepare.cmake~.
