Description: Add abi-version to library name 
Author: Gert Wollny <gw.fossdev@gmail.com>

--- a/CMake/dcmtkPrepare.cmake	2015-10-28 15:16:05.279429955 +0100
+++ b/CMake/dcmtkPrepare.cmake	2015-10-28 15:17:11.058822022 +0100
@@ -56,7 +56,7 @@
 #SET(DCMTK_PACKAGE_URL "http://www.dcmtk.org/")
 
 # Shared library version information
-SET(DCMTK_LIBRARY_PROPERTIES VERSION "${DCMTK_PACKAGE_VERSION}" SOVERSION "${DCMTK_ABI_VERSION}")
+SET(DCMTK_LIBRARY_PROPERTIES VERSION "${DCMTK_ABI_VERSION}.${DCMTK_PACKAGE_VERSION}" SOVERSION "${DCMTK_ABI_VERSION}")
 
 # General build options and settings
 OPTION(BUILD_APPS "Build command line applications and test programs." ON)
