diff -ru itksnap-2.4.0/CMake/find_fltk_13.cmake itksnap-2.4.0.new/CMake/find_fltk_13.cmake
--- itksnap-2.4.0/CMake/find_fltk_13.cmake	2012-11-21 17:50:38.000000000 +0100
+++ itksnap-2.4.0.new/CMake/find_fltk_13.cmake	2013-06-09 18:17:37.837308284 +0200
@@ -18,7 +18,7 @@
 SET(BASETARG "fltk${SUFF}")
 
 # Search for the base library
-FIND_LIBRARY(FLTK_BASE_LIBRARY NAMES ${BASETARG} PATHS "/usr/lib" "/usr/local/lib")
+FIND_LIBRARY(FLTK_BASE_LIBRARY NAMES ${BASETARG} PATHS "/usr/lib" "/usr/local/lib" "/usr/lib/fltk-1")
 
 IF(FLTK_BASE_LIBRARY)
   MESSAGE(STATUS "FLTK base: ${FLTK_BASE_LIBRARY}")
@@ -44,7 +44,7 @@
   # 2. In the root directory
   FIND_PATH(FLTK_INCLUDE_DIR
     NAMES "FL/Fl.H"
-    PATHS "${FLTK_ROOT_DIR}/include" "${FLTK_ROOT_DIR}")
+    PATHS "${FLTK_ROOT_DIR}/include" "/usr/include/fltk-1/" "${FLTK_ROOT_DIR}")
   MESSAGE(STATUS "FLTK include directory: ${FLTK_INCLUDE_DIR}")
 
   # For now, we let the user decide about the additional image libraries
Only in itksnap-2.4.0.new/CMake: find_fltk_13.cmake~
diff -ru itksnap-2.4.0/CMake/standalone.cmake itksnap-2.4.0.new/CMake/standalone.cmake
--- itksnap-2.4.0/CMake/standalone.cmake	2012-11-21 17:50:38.000000000 +0100
+++ itksnap-2.4.0.new/CMake/standalone.cmake	2013-06-09 18:12:42.021333899 +0200
@@ -7,7 +7,7 @@
 #############################################
 # REQUIRE VTK                               #
 #############################################
-FIND_PACKAGE(VTK 5.6 REQUIRED)
+FIND_PACKAGE(VTK REQUIRED)
 INCLUDE (${VTK_USE_FILE})
 
 #############################################
Only in itksnap-2.4.0.new/CMake: standalone.cmake~
diff -ru itksnap-2.4.0/Logic/Framework/IRISApplication.cxx itksnap-2.4.0.new/Logic/Framework/IRISApplication.cxx
--- itksnap-2.4.0/Logic/Framework/IRISApplication.cxx	2012-11-21 17:50:38.000000000 +0100
+++ itksnap-2.4.0.new/Logic/Framework/IRISApplication.cxx	2013-06-09 18:28:47.201250323 +0200
@@ -57,6 +57,7 @@
 #include "itkIdentityTransform.h"
 #include "itkResampleImageFilter.h"
 #include "itkNearestNeighborInterpolateImageFunction.h"
+#include "itkConstantBoundaryCondition.h"
 #include "itkBSplineInterpolateImageFunction.h"
 #include "itkLinearInterpolateImageFunction.h"
 #include "itkWindowedSincInterpolateImageFunction.h"
Only in itksnap-2.4.0.new/Logic/Framework: IRISApplication.cxx~
Only in itksnap-2.4.0.new/Logic/ImageWrapper: #GuidedNativeImageIO.cxx#
diff -ru itksnap-2.4.0/Logic/ImageWrapper/ScalarImageWrapper.txx itksnap-2.4.0.new/Logic/ImageWrapper/ScalarImageWrapper.txx
--- itksnap-2.4.0/Logic/ImageWrapper/ScalarImageWrapper.txx	2012-11-21 17:50:38.000000000 +0100
+++ itksnap-2.4.0.new/Logic/ImageWrapper/ScalarImageWrapper.txx	2013-06-09 18:29:44.157245391 +0200
@@ -20,6 +20,7 @@
 #include "itkImageSliceConstIteratorWithIndex.h"
 #include "itkNumericTraits.h"
 #include "itkRegionOfInterestImageFilter.h"
+#include "itkConstantBoundaryCondition.h"
 #include "itkRescaleIntensityImageFilter.h"
 #include "itkIdentityTransform.h"
 #include "itkResampleImageFilter.h"
Only in itksnap-2.4.0.new/Logic/ImageWrapper: ScalarImageWrapper.txx~
diff -ru itksnap-2.4.0/Logic/LevelSet/SNAPAdvectionFieldImageFilter.h itksnap-2.4.0.new/Logic/LevelSet/SNAPAdvectionFieldImageFilter.h
--- itksnap-2.4.0/Logic/LevelSet/SNAPAdvectionFieldImageFilter.h	2012-11-21 17:50:38.000000000 +0100
+++ itksnap-2.4.0.new/Logic/LevelSet/SNAPAdvectionFieldImageFilter.h	2013-06-09 18:26:54.952260043 +0200
@@ -50,8 +50,8 @@
     TInputImage,
     itk::Image<
       itk::CovariantVector<TOutputValueType,
-        ::itk::GetImageDimension<TInputImage>::ImageDimension>,
-      ::itk::GetImageDimension<TInputImage>::ImageDimension> >
+        TInputImage::ImageDimension>,
+	TInputImage::ImageDimension> >
 {
 public:
   
Only in itksnap-2.4.0.new/Logic/LevelSet: SNAPAdvectionFieldImageFilter.h~
