diff -ru InsightToolkit-4.8.2.orig/Modules/ThirdParty/DoubleConversion/CMakeLists.txt InsightToolkit-4.8.2/Modules/ThirdParty/DoubleConversion/CMakeLists.txt
--- InsightToolkit-4.8.2.orig/Modules/ThirdParty/DoubleConversion/CMakeLists.txt	2015-11-12 17:21:07.000000000 +0100
+++ InsightToolkit-4.8.2/Modules/ThirdParty/DoubleConversion/CMakeLists.txt	2015-12-03 22:00:01.771653734 +0100
@@ -7,11 +7,19 @@
 mark_as_advanced(ITK_USE_SYSTEM_DOUBLECONVERSION)
 
 if(ITK_USE_SYSTEM_DOUBLECONVERSION)
-  find_package(double-conversion NO_MODULE)
-  set(ITKDoubleConversion_SYSTEM_INCLUDE_DIRS
-    ${double-conversion_INCLUDE_DIRS})
-  set(ITKDoubleConversion_LIBRARIES
-    "${double-conversion_LIBRARIES}")
+  find_library(double-conversion_LIBRARIES double-conversion)
+  find_path(double-conversion_INCLUDE_DIRS double-conversion.h
+    PATH_SUFFIXES double-conversion
+    )
+
+  if (double-conversion_LIBRARIES AND double-conversion_INCLUDE_DIRS)
+    set(ITKDoubleConversion_SYSTEM_INCLUDE_DIRS
+      ${double-conversion_INCLUDE_DIRS})
+    set(ITKDoubleConversion_LIBRARIES
+      "${double-conversion_LIBRARIES}")
+  else()
+    MESSAGE(ERROR "double-conversion system library not found")
+  endif()
 else()
   set(ITKDoubleConversion_INCLUDE_DIRS
     ${ITKDoubleConversion_SOURCE_DIR}/src/double-conversion
