diff -ru insighttoolkit4-4.10.0.obsolete.0.73584495794876/Modules/ThirdParty/GIFTI/src/gifticlib/CMakeLists.txt insighttoolkit4-4.10.0/Modules/ThirdParty/GIFTI/src/gifticlib/CMakeLists.txt
--- insighttoolkit4-4.10.0.obsolete.0.73584495794876/Modules/ThirdParty/GIFTI/src/gifticlib/CMakeLists.txt	2016-05-26 17:32:22.000000000 +0200
+++ insighttoolkit4-4.10.0/Modules/ThirdParty/GIFTI/src/gifticlib/CMakeLists.txt	2016-07-06 16:10:33.255144401 +0200
@@ -22,8 +22,6 @@
   include_directories(${ITKNIFTI_INCLUDE_DIRS})
   set(PACKAGE_PREFIX "ITK")
   set(EXPAT_LIBRARIES ${ITKExpat_LIBRARIES})
-  set(ZLIB_LIBRARIES ITKznz)
-  set(NIFTI_LIBRARIES ITKniftiio)
   set(ZLIB_FOUND 1)
   set(GIFTI_LIBRARY_PROPERTIES ${ITK_LIBRARY_PROPERTIES})
 
Nur in insighttoolkit4-4.10.0/Modules/ThirdParty/GIFTI/src/gifticlib: CMakeLists.txt~.
diff -ru insighttoolkit4-4.10.0.obsolete.0.73584495794876/Modules/ThirdParty/NIFTI/CMakeLists.txt insighttoolkit4-4.10.0/Modules/ThirdParty/NIFTI/CMakeLists.txt
--- insighttoolkit4-4.10.0.obsolete.0.73584495794876/Modules/ThirdParty/NIFTI/CMakeLists.txt	2016-05-26 17:32:22.000000000 +0200
+++ insighttoolkit4-4.10.0/Modules/ThirdParty/NIFTI/CMakeLists.txt	2016-07-06 16:09:34.890864460 +0200
@@ -1,10 +1,9 @@
 project(ITKNIFTI)
-set(ITKNIFTI_THIRD_PARTY 1)
 
-set(ITKNIFTI_INCLUDE_DIRS
-  ${ITKNIFTI_SOURCE_DIR}/src/nifti/niftilib
-  ${ITKNIFTI_SOURCE_DIR}/src/nifti/znzlib
-  )
-set(ITKNIFTI_LIBRARIES ITKznz ITKniftiio)
+set(ITKNIFTI_SYSTEM_INCLUDE_DIRS "/usr/include/nifti")
+set(ZLIB_LIBRARIES znz)
+set(NIFTI_LIBRARIES niftiio)
+set(ITKNIFTI_LIBRARIES znz niftiio)
+set(ITKNIFTI_NO_SRC 1)
 
 itk_module_impl()
Nur in insighttoolkit4-4.10.0/Modules/ThirdParty/NIFTI: CMakeLists.txt~.
