--- insighttoolkit4-4.10.0.obsolete.0.73584495794876/Modules/Core/Common/include/itkMathDetail.h	2016-05-26 17:32:22.000000000 +0200
+++ insighttoolkit4-4.10.0/Modules/Core/Common/include/itkMathDetail.h	2016-07-06 15:58:30.820967520 +0200
@@ -28,7 +28,7 @@
 #ifndef itkMathDetail_h
 #define itkMathDetail_h
 
-#include "itkMath.h"
+//#include "itkMath.h"
 #include "itkIntTypes.h"
 #include "itkNumericTraits.h"
 
