--- gdcm-2.8.4/Utilities/doxygen/CMakeLists.txt	2017-11-09 09:09:58.000000000 +0100
+++ gdcm-2.8.4.new/Utilities/doxygen/CMakeLists.txt	2018-02-09 23:16:12.067080033 +0100
@@ -253,6 +253,7 @@
   set(XSLT_PROCESSOR ${LIBXSLT_XSLTPROC_EXECUTABLE})
 # http://docbook.sourceforge.net/release/xsl/current/doc/manpages/man.output.quietly.html
   set(XSLT_PROCESSOR_ARG --param man.output.quietly 1)
+  set(XSLT_PROCESSOR_ARG --maxdepth 5000)
   # User can change the behavior at cmake time:
   if(NOT DEFINED GDCM_MANPAGES_USE_NONET)
     set(GDCM_MANPAGES_USE_NONET FALSE)
