Index: dicom3tools_1.00.snapshot.20160402132920/config/Configure
===================================================================
--- dicom3tools_1.00.snapshot.20160402132920.orig/config/Configure
+++ dicom3tools_1.00.snapshot.20160402132920/config/Configure
@@ -354,14 +354,16 @@ then
 	if gccversion=`getVersion "$usegcc -v" | grep -v 'Configured with:' | grep -i version`
 	then
 		echo 1>&2 "Configure: using $usegcc version $gccversion"
-		if gplusplusversion=`getVersion "$usegpp -v" | grep -v 'Configured with:' | grep -i version`
+		if gplusplusversion=`getVersion "$usegpp --version"`
 		then
-			gplusplussubversion=`echo "$gplusplusversion" | sed -e 's/egcs-//' -e 's/Apple Computer.* based on //' -e 's/Apple LLVM version/LLVM version/'`
+
+			gplusplussubversion=`echo "$gplusplusversion" | grep g++`
 			gplusplusmajorversion=`echo "$gplusplussubversion" | awk '{print $3}' | sed 's/[.]/ /g' | awk '{print $1}'`
 			gplusplusminorversion=`echo "$gplusplussubversion" | awk '{print $3}' | sed 's/[.]/ /g' | awk '{print $2}'`
 			gplusplustinyversion=` echo "$gplusplussubversion" | awk '{print $3}' | sed 's/[.]/ /g' | awk '{print $3}'`
 
-			echo 1>&2 "Configure: using $usegpp version $gplusplusversion ($gplusplusmajorversion.$gplusplusminorversion.$gplusplustinyversion)"
+
+			echo 1>&2 "Configure: using ($gplusplusmajorversion.$gplusplusminorversion.$gplusplustinyversion)"
 
 			foundallgcc="yes"
 			echo >>generic.cf "#ifndef AnsiCCmd"
