Description: Use debian convention for pvrg naming
Author: Mathieu Malaterre <malat@debian.org>
Forwarded: not-needed
Last-Update: 2012-03-05

--- a/appsrc/dcfile/dcjpeg.script
+++ b/appsrc/dcfile/dcjpeg.script
@@ -13,7 +13,7 @@ DCCP=dccp
 DCTORAW=dctoraw
 DCKEY=dckey
 DCENCAP=dcencap
-JPEG=jpeg
+JPEG=pvrg-jpeg
 
 infile="$1"
 shift
--- a/appsrc/dcfile/dcunjpeg.script
+++ b/appsrc/dcfile/dcunjpeg.script
@@ -11,7 +11,7 @@ DCDUMP=dcdump
 DCTORAW=dctoraw
 DCKEY=dckey
 PNMTORAW=pnmtoraw
-JPEG=jpeg
+JPEG=pvrg-jpeg
 DJPEG=djpeg
 KDUEXPAND=kdu_expand
 
@@ -143,4 +143,4 @@ then
 else
 	echo 1>&2 "Could not find suitable codec or codec failed for Transfer Syntax $transfersyntaxuid"
 	exit 1
-fi
\ No newline at end of file
+fi
