Index: VTK-6.3.0/CMake/vtkModuleMacros.cmake
===================================================================
--- VTK-6.3.0.orig/CMake/vtkModuleMacros.cmake
+++ VTK-6.3.0/CMake/vtkModuleMacros.cmake
@@ -401,8 +401,8 @@ endfunction()
 function(vtk_target_name _name)
   get_property(_type TARGET ${_name} PROPERTY TYPE)
   if(NOT "${_type}" STREQUAL EXECUTABLE AND NOT VTK_JAVA_INSTALL)
-    set_property(TARGET ${_name} PROPERTY VERSION 1)
-    set_property(TARGET ${_name} PROPERTY SOVERSION 1)
+    set_property(TARGET ${_name} PROPERTY VERSION ${VTK_MAJOR_VERSION}.${VTK_MINOR_VERSION}.${VTK_BUILD_VERSION})
+    set_property(TARGET ${_name} PROPERTY SOVERSION ${VTK_MAJOR_VERSION}.${VTK_MINOR_VERSION})
   endif()
   if("${_name}" MATCHES "^[Vv][Tt][Kk]")
     set(_vtk "")
Index: VTK-6.3.0/Rendering/Tk/CMakeLists.txt
===================================================================
--- VTK-6.3.0.orig/Rendering/Tk/CMakeLists.txt
+++ VTK-6.3.0/Rendering/Tk/CMakeLists.txt
@@ -122,7 +122,7 @@ if(TK_FOUND AND VTK_USE_TK)
     vtk_target_install(vtkRenderingPythonTkWidgets)
 
     set_property(TARGET vtkRenderingPythonTkWidgets PROPERTY OUTPUT_NAME
-         vtkRenderingPythonTkWidgets-${VTK_MAJOR_VERSION}.${VTK_MINOR_VERSION})
+         vtkRenderingPythonTkWidgets)
 
     set(Module_PYTHON_LIBS ${VTK_TK_LIBRARIES})
     if(VTK_USE_X)
