commit 64d74b6c9adbf8d89551adde26e4c1576770f4c8
Author: Gert Wollny <gw.fossdev@gmail.com>
Date:   Wed Mar 15 21:11:54 2017 +0100

    Make it compile with VTK-7, Closes: #21

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c5f4e95..d27a59e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,7 +62,10 @@ SET_CUSTOM_wxWidgets_PACKAGE()
 ENDIF()
 
 IF(NOT USE_CUSTOM_VTK)
-FIND_PACKAGE(VTK 6.2.0 REQUIRED)
+  FIND_PACKAGE(VTK 7.0.0)
+  IF (NOT VTK_FOUND)
+    FIND_PACKAGE(VTK 6.2.0 REQUIRED)
+  ENDIF()
 ELSE()
 MESSAGE(STATUS "Using custom VTK")
 SET_CUSTOM_VTK_PACKAGE()
